roles/piwigo/tasks/main.yml

```diff
@@ -0,0 +1,40 @@
+---
+- name: Create Piwigo directories
+  file:
+    path: "{{ item }}"
+    state: directory
+  with_items:
+    - "{{ piwigo_data_directory }}/config"
+
+- name: Piwigo Mysql Docker Container
+  docker_container:
+    name: piwigo-mysql
+    image: mysql:5.7
+    pull: true
+    volumes:
+      - "{{ piwigo_data_directory }}/mysql:/var/lib/mysql:rw"
+    env:
+      MYSQL_DATABASE: "piwigo"
+      MYSQL_USER: "{{ piwigo_sql_user }}"
+      MYSQL_PASSWORD: "{{ piwigo_sql_password }}"
+      MYSQL_ROOT_PASSWORD: "{{ piwigo_sql_root_password }}"
+    restart_policy: unless-stopped
+
+- name: Piwigo Docker Container
+  docker_container:
+    name: piwigo
+    image: ghcr.io/linuxserver/piwigo
+    pull: true
+    volumes:
+      - "{{ piwigo_data_directory }}/config:/config:rw"
+      - "{{ piwigo_data_directory }}/pictures:/pictures"
+    ports:
+      - "{{ piwigo_port }}:8033"
+    restart_policy: unless-stopped
+    labels:
+      traefik.enable: "{{ piwigo_available_externally }}"
+      traefik.http.routers.piwigo.rule: "Host(`{{ piwigo_hostname }}.{{ ansible_nas_domain }}`)"
+      traefik.http.routers.piwigo.tls.certresolver: "letsencrypt"
+      traefik.http.routers.piwigo.tls.domains[0].main: "{{ ansible_nas_domain }}"
+      traefik.http.routers.piwigo.tls.domains[0].sans: "*.{{ ansible_nas_domain }}"
+      traefik.http.services.piwigo.loadbalancer.server.port: "8033"
```

Comment:
I added the following at the bottom, in line with labels, ports and restart_policy, so that at the install screen i can use db as host.
links:
  - piwigo-mysql:db